sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/indexes/models/FieldBuilderOptions.java

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.search.documents.indexes.models;
+
+import com.azure.core.util.serializer.MemberNameConverter;
+import com.azure.search.documents.indexes.SearchIndexAsyncClient;
+import com.azure.search.documents.indexes.SearchIndexClient;
+
+import java.util.Objects;
+
+/**
+ * Additional parameters to build {@link SearchField}.
+ */
+public class FieldBuilderOptions {
+    private MemberNameConverter serializer;
+
+    /**
+     * Gets the serializer use to build search fields in
+     * {@link SearchIndexClient#buildSearchField(Class, FieldBuilderOptions)} buildSearchField} or
+     * {@link SearchIndexAsyncClient#buildSearchField(Class, FieldBuilderOptions) buildSearchField}
+     *
+     * @return the custom serializer.
+     */
+    public MemberNameConverter getSerializer() {
+        return serializer;
+    }
+
+    /**
+     * Sets the custom serializer.
+     *
+     * @param serializer The custom serializer to set
+     * @return The {@link FieldBuilderOptions} object itself.
+     */
+    public FieldBuilderOptions setSerializer(MemberNameConverter serializer) {
+        this.serializer = Objects.requireNonNull(serializer, "The serializer cannot be null");
```

Comment:
I don't think this matches what the `FieldBuilder` API allows, that checks if the `serializer` is `null` and uses a default implementation.

Reply:
There is no necessary to use the API if no serializer to pass. They can simply pass new FieldBuilderOptions(). It is lazy initialization at the time we call static API

Reply:
I'm wrong, it checked if this type was `null`, not this property.